### PR TITLE
feat: create language endpoint

### PR DESCRIPTION
**Pull request by ZEED2468**
### Pull Request: Implement Endpoints to Get Languages

**Description:**
This PR introduces new endpoints in the `LanguagesController` for creating, retrieving, and updating languages. The endpoints include:

- An endpoint to retrieve the list of supported languages.

**Languages**
Languages

**Endpoints**

**2. Get Supported Languages**

- **Method:** GET
- **URL:** /api/v1/languages
- **Description:** Retrieves a list of all languages supported by the system.
- **Headers:**
  - Authorization: Bearer <token>
- **Success Response:**
  - **Status Code:** 200
  - **Body:**
    ```json
    {
      "status_code": 200,
      "message": "Languages fetched successfully",
      "languages": [
        {
          "language": "English (English)"
        },
        {
          "language": "Spanish (Español)"
        },
        {
          "language": "French (Français)"
        },
        {
          "language": "German (Deutsch)"
        },
        {
          "language": "Chinese (中文)"
        },
        {
          "language": "Japanese (日本語)"
        },
        {
          "language": "Russian (Русский)"
        },
        {
          "language": "Italian (Italiano)"
        },
        {
          "language": "Portuguese (Português)"
        },
        {
          "language": "Korean (한국어)"
        }
      ]
    }
    ```
- **Error Response (Unauthorized):**
  - **Status Code:** 401
    - **Body:**
      ```json
      {
        "status_code": 401,
        "message": "Unauthorized"
      }
      ```
LGET

**How to Test Manually**

**Create Language:**

- **Request Type:** POST
- **Endpoint:** /api/v1/languages
- **Request Body:** Include the language, code, and description fields.
- **Success Response:** Should return the created language with a status code of 201.
- **Error Responses:**
  - If the language already exists, should return a 409 status code with an appropriate error message.
  - If unauthorized or with an invalid token, should return a 401 status code with the appropriate error message.

**Fetch Languages:**

- **Request Type:** GET
- **Endpoint:** /api/v1/languages
- **Headers:** Include a valid Authorization token in the request headers.
- **Success Response:** Should return a list of supported languages with a status code of 200.
- **Error Response:** If unauthorized or with an invalid token, should return a 401 status code with the appropriate error message.

**Update Language:**

- **Request Type:** PATCH
- **Endpoint:** /api/v1/languages/{id}
- **Request Body:** Include the language, code, and description fields.
- **Success Response:** Should return the updated language with a status code of 200.
- **Error Responses:**
  - If the language does not exist, should return a 404 status code with an appropriate error message.
  - If unauthorized or with an invalid token, should return a 401 status code with the appropriate error message.

**Checklist**

- [x] Implemented the POST /api/v1/languages endpoint to create a new language.
- [x] Implemented the GET /api/v1/languages endpoint to retrieve the list of supported languages.
- [x] Implemented the PATCH /api/v1/languages/{id} endpoint to update an existing language.
- [x] Updated the LanguagesService to handle creating, retrieving, and updating languages.
- [x] Created unit tests for all three endpoints, including:
  - Testing successful creation, retrieval, and update of languages.
  - Handling errors for duplicate languages, non-existing languages, and unauthorized access.
- [x] Verified that the endpoints return the correct data format and status codes.
- [x] Added error handling for unauthorized access, existing languages, and non-existing languages.
- [x] Updated API documentation with new endpoint details and authentication requirements.

### CODE REVIEW

File: src/app.module.ts
Comment (Heba-WebDev) [inline src/app.module.ts:1]: Why is the NotificationsModule removed?

File: src/modules/languages/entities/language.entity.ts
Comment (Heba-WebDev) [inline src/modules/languages/entities/language.entity.ts:1]: Please specify the nullability for  for code, nullable: true/false

File: src/app.module.ts
Comment (ZEED2468) [reply src/app.module.ts:1]: Fixed already

**Review by Cobidex — DISMISSED** [review]

File: src/modules/languages/entities/language.entity.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';
+import { AbstractBaseEntity } from '../../../entities/base.entity';
+
+@Entity()
+export class Language extends AbstractBaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
```
Comment (Homoakin619) [inline src/modules/languages/entities/language.entity.ts:7]: remove the id definition here, the AbstractBaseEntity already took care of it

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]

**Pull request merged**